include/nil/crypto3/hash/detail/haifa_construction.hpp

```diff
@@ -61,7 +62,8 @@ namespace nil {
 
                 constexpr static const std::size_t digest_bits = Params::digest_bits;
                 constexpr static const std::size_t digest_bytes = digest_bits / octet_bits;
-                constexpr static const std::size_t digest_words = digest_bits / word_bits;
+                constexpr static const std::size_t digest_words =
```

Comment:
What?

Reply:
There was a problem during a digest conversion of truncated hashes to byte representation: namely, in the following line:

pack_n<endian_type, word_bits, octet_bits>(state_.data(), digest_words, d.data(), digest_bytes);

Pack_n checks first whether word_bits * digest_words == octet_bits * digest_bytes, and this check fails when digest_bits is not divisible by word_bits. In case of Blake2b, digest_bits can be, for example, equal to 224, while word_bits is equal to 64.

Currently this parameter is not used, although the proper value of digest_words should be calculated as ceil of digest_bits / word_bits.